src/pages/form/advanced-form/components/TableForm.tsx

```diff
@@ -0,0 +1,297 @@
+      editable: true,
+      isNew: true,
+    });
+    this.index += 1;
+    this.setState({ data: newData });
+  };
+
+  remove(key: string) {
+    const { data = [] } = this.state;
+    const { onChange } = this.props;
+    const newData = data.filter(item => item.key !== key);
+    this.setState({ data: newData });
+    if (onChange) {
+      onChange(newData);
+    }
+  }
+
+  handleKeyPress(e: React.KeyboardEvent, key: string) {
+    if (e.key === 'Enter') {
+      this.saveRow(e, key);
+    }
+  }
+
+  handleFieldChange(e: React.ChangeEvent<HTMLInputElement>, fieldName: string, key: string) {
+    const { data = [] } = this.state;
+    const newData = [...data];
+    const target = this.getRowByKey(key, newData);
+    if (target) {
+      target[fieldName] = e.target.value;
+      this.setState({ data: newData });
+    }
+  }
+
+  saveRow(e: React.MouseEvent | React.KeyboardEvent, key: string) {
+    e.persist();
+    this.setState({
+      loading: true,
+    });
+    setTimeout(() => {
+      if (this.clickedCancel) {
+        this.clickedCancel = false;
+        return;
+      }
+      const target = this.getRowByKey(key) || {};
+      if (!target.workId || !target.name || !target.department) {
+        message.error('请填写完整成员信息。');
+        (e.target as HTMLInputElement).focus();
+        this.setState({
+          loading: false,
+        });
+        return;
+      }
+      delete target.isNew;
+      this.toggleEditable(e, key);
+      const { data = [] } = this.state;
+      const { onChange } = this.props;
+      if (onChange) {
+        onChange(data);
+      }
+      this.setState({
+        loading: false,
+      });
+    }, 500);
+  }
+
+  cancel(e: React.MouseEvent, key: string) {
+    this.clickedCancel = true;
+    e.preventDefault();
+    const { data = [] } = this.state;
+    const newData = [...data];
+    // 编辑前的原始数据
+    let cacheOriginData = [];
+    cacheOriginData = newData.map(item => {
+      if (item.key === key) {
+        if (this.cacheOriginData[key]) {
+          const originItem = {
+            ...item,
+            ...this.cacheOriginData[key],
+            editable: false,
+          };
+          delete this.cacheOriginData[key];
+          return originItem;
+        }
+      }
+      return item;
+    });
+
+    this.setState({ data: cacheOriginData });
+    this.clickedCancel = false;
+  }
+
+  render() {
+    const { loading, data } = this.state;
+
+    return (
+      <Fragment>
+        <Table<TableFormDateType>
+          loading={loading}
+          columns={this.columns}
+          dataSource={data}
+          pagination={false}
+          rowClassName={record => (record.editable ? styles.editable : '')}
+        />
+        <Button
+          style={{ width: '100%', marginTop: 16, marginBottom: 8 }}
+          type="dashed"
+          onClick={this.newMember}
+        >
+          <PlusOutlined />
+          新增成员
+        </Button>
+      </Fragment>
+    );
+  }
+}
+
+export default TableForm;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of default exports is forbidden](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)